src/bson_mapper/test/bson_streambuf.cpp

```diff
@@ -29,17 +32,28 @@ class doc_validator_callback {
     doc_validator_callback(bsoncxx::document::value doc) : _doc(doc) {
     }
 
+    // Getter for the number of document compared so far.
+    int count() {
+        return _count;
+    }
+
     /**
      * This asserts that the given document is the same as the reference document.
      */
     void operator()(bsoncxx::document::value v) {
+        _count++;
         REQUIRE(v.view() == _doc.view());
     }
 
    private:
+    // Keeps track of the number of documents compared by this validator.
+    static int _count;
```

Comment:
When I made it non-static, I ran into problems with the callback object being copied when passed into the streambuffer.

 I could make bson_streambuf accept a pointer to a document_callback, or have a member function bson_streambuf::callback() that returns such a pointer.